plugins/rbac/src/components/CreateRole/RoleDetailsForm.tsx

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+
+import { TextField } from '@material-ui/core';
+
+type RoleDetailsFormProps = {
+  name: string;
+  description?: string;
+  nameError?: string;
+  handleBlur: React.FocusEventHandler<HTMLInputElement | HTMLTextAreaElement>;
+  handleChange: React.ChangeEventHandler<
+    HTMLTextAreaElement | HTMLInputElement
+  >;
+};
+
+export const RoleDetailsForm = ({
+  name,
+  description,
+  nameError,
+  handleBlur,
+  handleChange,
+}: RoleDetailsFormProps) => {
+  return (
+    <div
+      style={{
+        display: 'flex',
+        flexDirection: 'column',
+        gap: '20px',
+      }}
+    >
+      <TextField
+        required
+        label="Name"
+        variant="outlined"
+        id="role-name"
+        data-testid="role-name"
+        aria-labelledby="name"
+        helperText={nameError ?? 'Enter name of the role'}
+        value={name}
+        name="name"
+        onChange={handleChange}
+        onBlur={handleBlur}
+        error={!!nameError}
+      />
+      <TextField
+        label="Description"
+        variant="outlined"
+        helperText="Enter a brief description about the role(The purpose of the role)"
```

Comment:
Nit: this text is per UXD?

```suggestion
        helperText="Enter a brief description about the role (The purpose of the role)"
```

Reply:
@invincibleJai Yes it is as per https://www.figma.com/file/E6C9X3OSVzlWkW7ZsbxN7s/Administration?type=design&node-id=932-1542&mode=design&t=V40DvGtSyENzJFNB-4.

Reply:
so there is space :)

Reply:
yes, added the space.